### ISSUE TITLE
[BUG] Components from motion.custom doesn't animate

### ISSUE THREAD

**Issue by AlexanderMelox** [framer/motion#853]
# Describe
When using a component created from `motion.custom(Component)`, it totally breaks the component and doesn't do any motion. You can't use variants, initial nor animate. 

## Reproduce

https://codesandbox.io/s/boring-glade-742qs?file=/src/App.js

1. Click on codesandbox link
2. Click on refresh and you can see that the `MotionDiv` coming from a `motion.custom()` does not seem to work. 

**7. Environment details**

Google chrome, Mac OS. Framer motion version: 2.9.4

**Comment by AlexFlasch** [comment]
I believe that framer-motion needs to access the DOM in order to animate values, so it requires a ref to be present on `motion.custom()`.   
The console error in your sandbox seems to suggest that framer-motion is trying to do just that, but refs can't be placed directly on functional components. The solution I found was to wrap your functional component in `React.forwardRef` so that the library can do its thing!

I changed:
```javascript
const Div = ({ children }) => <div>{children}</div>;
```
to:
```javascript
const Div = React.forwardRef(({ children }, ref) => <div ref={ref}>{children}</div>);
```
and your example works for me now!

Without better knowledge of framer-motion's inner workings I can't say whether this is the intended / ideal solution though.
